File: fizzbuzz/scala/fizzbuzz.scala
object fizzbuzz {

  def fizzbuzz(n:Int) = {
    println((n % 3, n % 5) match {
      case (0, 0) => "FizzBuzz"
      case (0, _) => "Fizz"
      case (_, 0) => "Buzz"
      case _ => n
    })
  }

  def main(args: Array[String]) {
    (1 until 100).map(fizzbuzz _)
  }
}

File: fizzbuzz/rust/fizzbuzz.rs
fn fizzbuzz(n: i32) {
    for x in 1..n + 1 {
        match (x % 3, x % 5) {
            (0, 0) => println!("FizzBuzz"),
            (0, _) => println!("Fizz"),
            (_, 0) => println!("Buzz"),
            _ => println!("{}", x),
        }
    }
}

fn main() {
    fizzbuzz(100);
}
